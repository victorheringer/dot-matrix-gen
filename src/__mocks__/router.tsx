import { MemoryRouter } from "react-router-dom";

export function RouterWrapper({ children }: any) {
  return <MemoryRouter>{children}</MemoryRouter>;
}
