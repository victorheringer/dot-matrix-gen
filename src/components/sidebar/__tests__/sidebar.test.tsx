import { render } from "@testing-library/react";
import { Sidebar } from "../index";
import { RouterWrapper } from "../../../__mocks__/router";

describe("Sidebar", () => {
  it("should render correctly", () => {
    const tree = render(
      <RouterWrapper>
        <Sidebar />
      </RouterWrapper>
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });
});
