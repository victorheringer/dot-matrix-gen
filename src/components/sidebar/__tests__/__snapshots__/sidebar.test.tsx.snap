// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ iEichk"
  >
    <h1
      class="sc-gsnTZi gUxsMH"
    >
      Dot Matrix Gen
    </h1>
    <ul
      class="sc-dkzDqf iuqTVU"
    >
      <li
        class="sc-hKMtZM dKFOdX"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="sc-hKMtZM dKFOdX"
      >
        <a
          href="/projects"
        >
          Projects
        </a>
      </li>
      <li
        class="sc-hKMtZM dKFOdX"
      >
        <a
          href="/settings"
        >
          Settings
        </a>
      </li>
      <li
        class="sc-hKMtZM dKFOdX"
      >
        <a
          href="https://github.com/victorheringer/dot-matrix-gen"
          target="_blank"
        >
          Github
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
